File: server/app/models/llm_config.py
from sqlalchemy import Column, Integer, String, DateTime, Text
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base
from .base import LOCALTIME


class LLMConfig(Base):
    __tablename__ = "llm_configs"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    model_id = Column(String(100), nullable=False)
    base_url = Column(String(255), nullable=False)
    api_key = Column(String(255), nullable=False)
    description = Column(Text, nullable=False, default='')
    created_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, nullable=False)
    updated_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, onupdate=datetime.now, nullable=False)

    agents = relationship(
        "Agent",
        back_populates="llm_config",
        primaryjoin="LLMConfig.id == Agent.llm_config_id",
        foreign_keys="Agent.llm_config_id"
    )

File: server/internal/model/llm_config.go
package model

import "time"

// LLMConfig stores the configuration for connecting to an LLM API provider.
// Includes model ID, base URL, and API key for the OpenAI-compatible API.
type LLMConfig struct {
	ID          int64     `gorm:"primaryKey;autoIncrement;type:INTEGER PRIMARY KEY AUTOINCREMENT" json:"id"`
	Name        string    `gorm:"type:varchar(100);not null" json:"name"`
	ModelID     string    `gorm:"type:varchar(100);not null;column:model_id" json:"model_id"`
	BaseURL     string    `gorm:"type:varchar(255);not null;column:base_url" json:"base_url"`
	APIKey      string    `gorm:"type:varchar(255);not null;column:api_key" json:"api_key"`
	Description string    `gorm:"type:text;not null;default:''" json:"description"`
	CreatedAt   time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
	UpdatedAt   time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (LLMConfig) TableName() string { return "llm_configs" }
